libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Util/SFSDKOAuth2.h

```diff
@@ -0,0 +1,116 @@
+ FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+ CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+ WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY
+ WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#import <Foundation/Foundation.h>
+/** SFOAuth default network timeout in seconds.
+ */
+extern const NSTimeInterval kSFOAuthDefaultTimeout;
+
+/** This constant defines the SFOAuth framework error domain.
+ 
+ Domain indicating an error occurred during OAuth authentication.
+ */
+extern NSString * _Nonnull const kSFOAuthErrorDomain;
+
+/**
+ @enum SFOAuthErrorDomain related error codes
+ Constants used by SFOAuthCoordinator to indicate errors in the SFOAuth domain
+ */
+enum {
+    kSFOAuthErrorUnknown = 666,
+    kSFOAuthErrorTimeout,
+    kSFOAuthErrorMalformed,
+    kSFOAuthErrorAccessDenied,              // end user denied authorization
+    kSFOAuthErrorInvalidClientId,
+    kSFOAuthErrorInvalidClientCredentials,  // client secret invalid
+    kSFOAuthErrorInvalidGrant,              // expired access/refresh token, or IP restricted, or invalid login hours
+    kSFOAuthErrorInvalidRequest,
+    kSFOAuthErrorInactiveUser,
+    kSFOAuthErrorInactiveOrg,
+    kSFOAuthErrorRateLimitExceeded,
+    kSFOAuthErrorUnsupportedResponseType,
+    kSFOAuthErrorWrongVersion,              // credentials do not match current Connected App version in the org
+    kSFOAuthErrorBrowserLaunchFailed,
+    kSFOAuthErrorUnknownAdvancedAuthConfig,
+    kSFOAuthErrorInvalidMDMConfiguration,
+    kSFOAuthErrorJWTInvalidGrant
+};
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SFSDKOAuthTokenEndpointErrorResponse : NSObject
+@property  (nonatomic, readonly) NSString *tokenEndpointErrorCode;
+@property  (nonatomic, readonly) NSString *tokenEndpointErrorDescription;
+@property  (nonatomic, readonly) NSError *error;
+@end
+
+
+@interface SFSDKOAuthTokenEndpointRequest : NSObject
+@property (nonatomic, copy, nullable) NSString *refreshToken;
+@property (nonatomic, copy, nullable) NSString *userAgentForAuth;
+@property (nonatomic, copy, nullable) NSString *redirectURI;
+@property (nonatomic, copy, nullable) NSString *clientID;
+@property (nonatomic, copy, nullable) NSString *approvalCode;
+@property (nonatomic, copy, nullable) NSString *codeVerifier;
+@property (nonatomic, assign) NSTimeInterval timeout;
+@property (nonatomic, strong, nullable) NSURL *serverURL;
+@property (nonatomic, strong, nullable) NSDictionary * additionalTokenRefreshParams;
+@property (nonatomic, strong, nullable) NSArray<NSString *> *additionalOAuthParameterKeys;
+@end
+
+@interface SFSDKOAuthTokenEndpointResponse : NSObject
```

Comment:
Is this a redefined interface from the one above? Can we combine them, since most properties are `nullable` anyway?

Reply:
My bad - I didn't read the `Request` and `Response` difference ;-) Ignore this comment.